test/e2e_node/topology_manager_test.go

```diff
@@ -414,38 +415,47 @@ func waitForAllContainerRemoval(podName, podNS string) {
 }
 
 func runTopologyManagerPositiveTest(f *framework.Framework, numPods int, ctnAttrs, initCtnAttrs []tmCtnAttribute, envInfo *testEnvInfo) {
-	var pods []*v1.Pod
+	podMap := make(map[string]*v1.Pod)
 
 	for podID := 0; podID < numPods; podID++ {
 		podName := fmt.Sprintf("gu-pod-%d", podID)
 		framework.Logf("creating pod %s attrs %v", podName, ctnAttrs)
 		pod := makeTopologyManagerTestPod(podName, ctnAttrs, initCtnAttrs)
 		pod = f.PodClient().CreateSync(pod)
 		framework.Logf("created pod %s", podName)
-		pods = append(pods, pod)
+		podMap[podName] = pod
 	}
 
 	// per https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/693-topology-manager/README.md#multi-numa-systems-tests
 	// we can do a menaingful validation only when using the single-numa node policy
 	if envInfo.policy == topologymanager.PolicySingleNumaNode {
-		for podID := 0; podID < numPods; podID++ {
-			validatePodAlignment(f, pods[podID], envInfo)
+		for _, pod := range podMap {
+			validatePodAlignment(f, pod, envInfo)
 		}
 		if envInfo.scope == podScopeTopology {
-			for podID := 0; podID < numPods; podID++ {
-				err := validatePodAlignmentWithPodScope(f, pods[podID], envInfo)
+			for _, pod := range podMap {
+				err := validatePodAlignmentWithPodScope(f, pod, envInfo)
 				framework.ExpectNoError(err)
 			}
 		}
 	}
 
-	for podID := 0; podID < numPods; podID++ {
-		pod := pods[podID]
-		framework.Logf("deleting the pod %s/%s and waiting for container removal",
-			pod.Namespace, pod.Name)
-		deletePodSyncByName(f, pod.Name)
-		waitForAllContainerRemoval(pod.Name, pod.Namespace)
+	deletePodsAsync(f, podMap)
+}
+
+func deletePodsAsync(f *framework.Framework, podMap map[string]*v1.Pod) {
+	var wg sync.WaitGroup
+	for _, pod := range podMap {
```

Comment:
some stupid question, why do you get under arguments `map[string]*v1.Pod` instead of `[]*v1.Pod`, when you do not really use the key?

Reply:
It's not stupid: I want to use this helper from two different test suites.
One of them (podresources) already keeps a map, while here the most natural representation would be a `[]*v1.Pod`. So I need to refactor at least one between the two calling sites and the code :)
Now: the code originates from `podresources_test.go`, and I want to use here; to minimize the changes the path of least resistance is to use a map here in `topology_manager_test.go`.
I'm open to other approaches though.

Reply:
good enough for me.